tests/Sulu/Component/Content/Document/Query/QueryBuilderFunctionalTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Sulu\Component\Content\Document\Query;
+
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+use Sulu\Component\Content\Document\WorkflowStage;
+
+class QueryBuilderFunctionalTest extends SuluTestCase
+{
+    public function setUp()
+    {
+        $this->manager = $this->getContainer()->get('sulu_document_manager.document_manager');
+    }
+
+    /**
+     * It should be able to use a structures fields as criteria.
+     */
+    public function testStructureFieldsCriteria()
+    {
+        $builder = $this->manager->createQueryBuilder();
+        $builder
+            ->setLocale('de')
+            ->useStructure('p', 'overview')
+            ->from()->document('page', 'p')->end()
+            ->where()->eq()->field('p.structure#article')->literal('hello');
```

Comment:
@wachterjohannes @danrot @chirimoya This is the syntax for selecting from structures atm. I wanted to use dot notation instead of `#`, but the ODM query builder validates this. I think `#` is a good compromise.